devito/symbolics/extended_sympy.py

```diff
@@ -262,6 +308,8 @@ def __new__(cls, params):
         for p in as_tuple(params):
             if isinstance(p, str):
                 args.append(Symbol(p))
+            elif isinstance(p, int):
+                args.append(Integer(p))
```

Comment:
I assume you've checked already that if we don't cast to Integer sympy complains and raises an exception?

Reply:
If we don't cast to `Integer` this constructor complains

Reply:
It doesn't complain but it seemed cleaner to use sympy types everywhere

Reply:
they should be converted internally to Integer

Reply:
:+1: